File: 1. C++/03. Data Structures/Linear/linked_list_stack_queue.cpp
#include <iostream>
#include <queue>
#include <stack>
using namespace std;

struct Node {
    int data;
    Node* next;

    explicit Node(int value) : data(value), next(nullptr) {}
};

void printList(Node* head) {
    while (head != nullptr) {
        cout << head->data << ' ';
        head = head->next;
    }
    cout << "\n";
}

int main() {
    Node* head = new Node(10);
    head->next = new Node(20);
    head->next->next = new Node(30);

    cout << "Linked list: ";
    printList(head);

    stack<int> st;
    st.push(5);
    st.push(15);
    cout << "Stack top: " << st.top() << "\n";
    st.pop();

    queue<int> q;
    q.push(1);
    q.push(2);
    cout << "Queue front: " << q.front() << "\n";

    while (head != nullptr) {
        Node* nextNode = head->next;
        delete head;
        head = nextNode;
    }

    return 0;
}

File: 3. Python/03. Data Structures/Linear/linked_list_stack_queue.py
from collections import deque


class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


def print_list(head):
    current = head
    while current:
        print(current.data, end=" ")
        current = current.next
    print()


head = Node(10)
head.next = Node(20)
head.next.next = Node(30)
print_list(head)

stack = []
stack.append(5)
stack.append(15)
print("Stack top:", stack[-1])

queue = deque([1, 2])
print("Queue front:", queue[0])
